src/integrationTest/sam.test.ts

```diff
@@ -53,153 +85,349 @@ async function getSamCodeLenses(documentUri: vscode.Uri): Promise<vscode.CodeLen
             if (!codeLenses || codeLenses.length === 0) {
                 continue
             }
+
             // omnisharp spits out some undefined code lenses for some reason, we filter them because they are
             // not shown to the user and do not affect how our extension is working
-            codeLenses = codeLenses.filter(lens => lens !== undefined && lens.command !== undefined)
-            if (codeLenses.length === 3) {
-                return codeLenses as vscode.CodeLens[]
+            codeLenses = codeLenses.filter(codeLens => {
+                if (codeLens.command && codeLens.command.arguments && codeLens.command.arguments.length === 1) {
+                    // tslint:disable: no-unsafe-any
+                    return (
+                        codeLens.command.command === getInvokeCmdKey(language) &&
+                        codeLens.command.arguments[0].isDebug === debug
+                    )
+                    // tslint:enable: no-unsafe-any
```

Comment:
this is as many lines as putting disable above each line in the return, would prefer that since it's easy to accidentally delete the enable